pkg/ottl/boolean_value.go

```diff
@@ -8,195 +8,280 @@ import (
 	"fmt"
 )
 
-// boolExpressionEvaluator is a function that returns the result.
-type boolExpressionEvaluator[K any] func(ctx context.Context, tCtx K) (bool, error)
-
 // BoolExpr represents a condition in OTTL
-type BoolExpr[K any] struct {
-	boolExpressionEvaluator[K]
+type BoolExpr[K any] interface {
```

Comment:
Let's make this private, it looks like there aren't any usages outside OTTL right now. We'll need a breaking API changelog entry.